WordPress/Classes/ViewRelated/WhatsNew/Data store/AnnouncementsStore.swift

```diff
@@ -121,18 +121,22 @@ class CachedAnnouncementsStore: AnnouncementsStore {
 }
 
 
-private extension CachedAnnouncementsStore {
+extension CachedAnnouncementsStore {
 
     func cacheIsValid(for announcements: [Announcement]) -> Bool {
-        guard let minimumVersion = announcements.first?.minimumAppVersion,   // there should not be more than one announcement
-              let maximumVersion = announcements.first?.maximumAppVersion,   // per version, but if there is, each of them must match the version
-              let targetVersions = announcements.first?.appVersionTargets,   // so we might as well choose the first
-              let version = versionProvider.version,
-              ((minimumVersion...maximumVersion).contains(version) || targetVersions.contains(version)) else {
+        // There should not be more than one announcement per version,
+        // but if there is, each of them must match the version,
+        // so we might as well choose the first.
+        guard let announcement = announcements.first, let version = versionProvider.version else {
             return false
         }
-        return true
+
+        return (announcement.minimumAppVersion...announcement.maximumAppVersion).contains(version)
+        || announcement.appVersionTargets.contains(version)
     }
+}
```

Comment:
I don't feel bad about making this method `internal` over `private` "just" to test it.  The logic looks simple but, as the test shows, has various possible paths, so testing it thoroughly is valuable.

One option I considered was to extract it in a dedicated object, which would have made it less awkward to have the logic accessible by all `internal` consumers of `CachedAnnouncementsStore`. I didn't go down that route because the context of this PR was merely to upgrade SwiftLint.